src/main/java/com/google/codeu/data/Eventstore.java

```diff
@@ -0,0 +1,76 @@
+package com.google.codeu.data;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.FilterOperator;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.UUID;
+import java.util.Set;
+import java.util.HashSet;
+
+/** Provides access to the data stored in eventstore. */
```

Comment:
I think you can use the Datastore for this purpose, and there is probably no need to make a separate Evenstore. The fact that you are storing Entity that has a "Event" as its first parameter to its constructor (line 31) make us able to store event entities separate as message entities, or user entities. So I think we can query events, messages, and users using the same Datastore (will help us keep the same functionality in the same place  maybe!). If I am missing something here, and there is actually a need for Eventstore, or if for some reason having it separate is better I would be happy to know! that is all a suggestion :)